btsieve/src/main.rs

```diff
@@ -201,48 +202,42 @@ fn create_ethereum_routes(
 
             let web3_client = web3_client.clone();
 
-            let blocks = ethereum_block_listener(
-                web3_client.clone(),
-                Duration::from_secs(match network {
-                    EthereumNetwork::Mainnet => 5,
-                    EthereumNetwork::Ropsten => 5,
-                    EthereumNetwork::Regtest => 1,
-                    EthereumNetwork::Unknown => 1,
-                }),
-            )
-            .expect("Should return a Web3 block poller");
+            let block_source = runtime.block_on(Web3HttpBlockSource::new(web3_client.clone()))?;
 
             let executor = runtime.executor();
-            let web3_processor = blocks.for_each(move |block| {
-                ethereum_chain.add_block(block.clone());
-                ethereum::check_block_queries(block_query_repository.clone(), block.clone())
-                    .for_each(|QueryMatch(id, block_id)| {
-                        block_query_result_repository.add_result(id.0, block_id);
+            let web3_processor = block_source
+                .blocks()
+                .map_err(|e| log::warn!(target: "ethereum", "error fetching latest block {:?}", e))
```

Comment:
The implication of this design is that if the block_source error bubbles up then it becomes irrecoverable as the polling would stop (I think).

It may be interesting to point that out in the log or in the Trait implementation.

Reply:
Not bubbeling up IO errors now and added a test that verifies that.